src/linked_list.py

```diff
@@ -47,39 +47,36 @@ def pop(self):
             self.header = pop_node.pointer
             self.length -= 1
             return pop_node.value
+        else:
+            raise IndexError('Cannot pop from an empty list.')
 
     def display(self):
         """Return a unicode string representing the list."""
         if self.length:
+            return_str = '{0}{1}{2}'.format('(', self.header.value, ', ')
             current_node = self.header
-            return_str = '('
             while current_node.pointer:
-                my_value = current_node.value
+                my_value = current_node.pointer.value
```

Comment:
Using `current_node.pointer` instead of `current_node` makes me uncomfortable but it works.